src/App.js

```diff
@@ -1,17 +1,46 @@
-import { Routes } from 'react-router-dom';
+import { Routes, Route, useLocation } from 'react-router-dom';
 import './App.css';
+import { useEffect, useState } from "react";
+import { getTopics } from "./utils/api";
+import{ getLocation } from "./utils/helpers"
+import Breadcrumbs from './components/Breadcrumbs';
 import Header from "./components/Header";
-import QueryForm from "./components/QueryForm";
-import Articles from './components/Articles';
+import Home from './components/Home';
+import Topic from "./components/Topic";
 
 function App() {
-  return <div>
-        <p id="breadcrumbs">Breadcrumbs...</p>
-        <Header/>
-        <QueryForm/>
-        <Articles/>
-        <Routes></Routes>
-    </div>
+
+  const location = useLocation();
+  const [topicSlug, setTopicSlug] = useState(getLocation(location));
+
+  const [topicsList, setTopicsList] = useState([]);
+
+  const [queryState, setQuerystate] = useState("");
+
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+      getTopics()
+          .then((topicsFromApi) => {
+              setTopicsList(topicsFromApi);
+              setIsLoading(false)
+          })
+  }, [])
+
+if (isLoading) {
+  return <div className="loading">
+            <p>Getting some data for you...</p>
+        </div>
+}
+
+return <div>
```

Comment:
I would put this in an else block :)

Reply:
Or I will once I've reviewed your other comment, my bad lol